community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java

```diff
@@ -254,16 +254,17 @@ private FilenameFilter getNativeIndexFileFilter( boolean inverse )
         return ( dir, name ) ->
         {
             File file = new File( dir, name );
-            if ( outsideWorld.fileSystem().isDirectory( file ) )
-            {
-                // Always go down directories
-                return true;
-            }
-            if ( name.equals( FailureStorage.DEFAULT_FAILURE_FILE_NAME ) )
-            {
-                // Never include failure-storage files
-                return false;
-            }
+            // Lucene index files lives in:
+            // - schema/index/lucene/<indexId>/<partition>/.....
+            // - schema/index/lucene_native-x.y/<indexId>/lucene-x.y/x/.....
+            return !outsideWorld.fileSystem().isDirectory(file) || ((!file.toPath().getParent().equals(baseSchemaIndexPath) || !name.equals("lucene")) &&
```

Comment:
This class facing compilation problems now and also i wouldn't say that this condition is readable at all in its current state